### PR TITLE
Major Refactor

### PR DESCRIPTION
**Pull request by dahanj95**
Reorganized the code

### CODE REVIEW

**Review by thewh1teagle — CHANGES_REQUESTED** [review]

File: src/watchdog.py [hunk 1]
```diff
@@ -1,59 +1,27 @@
...
-
-
-    def bypass_incapsula(self):
-        self.s.headers = {
+
+class watchdog:
+
+    def __init__(self, branch, month, onAvailable) -> None:
+
+        self.branch = branch
```
[the remark this answers is not in the document]
Comment (dahanj95) [reply src/watchdog.py:13]: Yakov, branch in english means "sniff" סניף

File: .env [hunk 2]
```diff
@@ -0,0 +1,2 @@
+branch=11
```
[the remark this answers is not in the document]
Comment (dahanj95) [reply .env:1]: I removed in my commit settings.py and migrated all settings to .env file

File: src/main.py [hunk 3]
```diff
@@ -1,30 +1,35 @@
...
-                time.sleep(60*30) # 30 minutes
-    except Exception as e:
-        print(f'We got error :( {e}')
-        push.send(settings.TOPIC, "Error", e, "")
-
-if __name__ == '__main__':  
+
+
+# this is a callback that gets called when there's an appointment available
+def onAvailable(cnt, cname):
```
[the remark this answers is not in the document]
Comment (dahanj95) [reply src/main.py:10]: Naming conventions, I personally name like i'm used to in .net. You can change it to snake case and merge the pull.

File: src/watchdog.py [hunk 4]
```diff
@@ -1,59 +1,27 @@
...
-        self.s = requests.session()
-        self.bypass_incapsula()
-
-
-    def bypass_incapsula(self):
-        self.s.headers = {
+
+class watchdog:
+
+    def __init__(self, branch, month, onAvailable) -> None:
```
[the remark this answers is not in the document]
Comment (dahanj95) [reply src/watchdog.py:11]: Yep, if you want i can add support to multiple dates and locations in the future. Do you need it now?

File: src/watchdog.py [hunk 5]
```diff
@@ -89,18 +61,23 @@ def available_appointment(self, location: Location, month) -> True:
...
         )
-        data = response.json()
-        return len(data) if isinstance(data, list) and len(data) > 0 else 0
\ No newline at end of file
+
+        arrJson = response.json()
+
+        available = len(arrJson) if isinstance(
+            arrJson, list) and len(arrJson) > 0 else 0
+        if (available > 0):
```
[the remark this answers is not in the document]
Comment (dahanj95) [reply src/watchdog.py:82]: Still learning from you everyday :)

File: src/watchdog.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dahanj95) [reply src/watchdog.py:13]: Do you still want to replace "branch" with a different term? I think branch is suitable, I'll wait for you to decide and reply back.

File: src/watchdog.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (dahanj95) [reply src/watchdog.py]: Working on a structural change to add support for multiple months and branches (locations).

File: src/watchdog.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (thewh1teagle) [reply src/watchdog.py:13]: Got it. branch is better name ok

**Comment by dahanj95** [comment]
Okay Yakov, I've done everything you mentioned ;] I think it's safe to merge the pull request.

**Pull request closed without merging**